teos/src/responder.rs

```diff
@@ -246,88 +247,60 @@ impl Responder {
     ) {
         let tracker = TransactionTracker::new(breach, user_id, status);
 
-        self.trackers
-            .lock()
-            .unwrap()
-            .insert(uuid, tracker.get_summary());
-
-        let mut tx_tracker_map = self.tx_tracker_map.lock().unwrap();
-        if let Some(map) = tx_tracker_map.get_mut(&tracker.penalty_tx.txid()) {
-            map.insert(uuid);
+        if let Ok(_) = self.dbm.lock().unwrap().store_tracker(uuid, &tracker) {
+            log::info!("New tracker added (uuid={uuid})");
         } else {
-            tx_tracker_map.insert(tracker.penalty_tx.txid(), HashSet::from_iter(vec![uuid]));
+            log::error!(
+                "Failed to store tracker in database (uuid={uuid}). It might be already stored."
+            );
         }
-
-        self.dbm
-            .lock()
-            .unwrap()
-            .store_tracker(uuid, &tracker)
-            .unwrap();
-        log::info!("New tracker added (uuid={uuid})");
     }
 
     /// Checks whether a given tracker can be found in the [Responder].
     pub(crate) fn has_tracker(&self, uuid: UUID) -> bool {
         // has_tracker should return true as long as the given tracker is hold by the Responder.
         // If the tracker is partially kept, the function will log and the return will be false.
         // This may point out that some partial data deletion is happening, which must be fixed.
-        self.trackers
-            .lock()
-            .unwrap()
-            .get(&uuid)
-            .map_or(false, |tracker| {
-                self.tx_tracker_map
-                    .lock()
-                    .unwrap()
-                    .get(&tracker.penalty_txid)
-                    .map_or(
-                        {
-                            log::debug!(
-                            "Partially found Tracker. Some data may have not been properly deleted"
-                        );
-                            false
-                        },
-                        |_| true,
-                    )
-            })
+
+        // TODO: Use another DBM method that doesn't de-ser the whole tracker.
+        // TODO: Consider removing this method (has_tracker) all together.
```

Comment:
I don't follow

Reply:
The first says we should make a DB method that checks only on UUID without de serializing a whole tracker.
The second says has_tracker is actually useless, but if you ask me how, i won't recall :joy:
I had it in mind when I was thinking how to refactor `Watcher::add_appointment` :(